### PR TITLE
[KIWI-1567] - CIC | FE | Accessibility Audit | Format dates on CYA

### PR DESCRIPTION
**Pull request by GDainton**
### What changed

Changed formatDate function to format the user's date of birth in the check answers page as DD Month YYYY, where month is either in English or Welsh depending on the language cookie.

### Why did it change

To reflect the outcome of the WCAG accessibility audit.

### Issue tracking

- [KIWI-1567](https://govukverify.atlassian.net/browse/KIWI-1567)

Screenshot 2024-02-14 at 15 13 09

Screenshot 2024-02-14 at 15 12 56

Screenshot 2024-02-14 at 15 12 17

Screenshot 2024-02-14 at 15 12 06

[KIWI-1567]: https://govukverify.atlassian.net/browse/KIWI-1567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,8 +28,8 @@ COPY --from=builder /app/yarn.lock ./
 COPY --from=builder /app/src ./src
 
 # Add in dynatrace layer
-COPY --from=khw46367.live.dynatrace.com/linux/oneagent-codemodules-musl:nodejs / /
-ENV LD_PRELOAD /opt/dynatrace/oneagent/agent/lib64/liboneagentproc.so
+# COPY --from=khw46367.live.dynatrace.com/linux/oneagent-codemodules-musl:nodejs / /
+# ENV LD_PRELOAD /opt/dynatrace/oneagent/agent/lib64/liboneagentproc.so
```
Comment (Caitlin-cooling) [inline Dockerfile:32]: Please undo

File: src/app/cic/controllers/checkDetails.js [hunk 2]
```diff
@@ -16,11 +16,12 @@ class CheckDetailsController extends DateController {
       const changeUrl = req.sessionModel.get("changeUrl");
       const firstName = req.sessionModel.get("firstName");
       const middleName = req.sessionModel.get("middleName");
-      const surname = req.sessionModel.get("surname");
-      const fullName = firstName + " " + middleName + " " + surname;
-
+      const surname = req.sessionModel.get("surname")
+      const fullName = firstName + " " + middleName + " " + surname
+      const language = req.lng
```
Comment (Caitlin-cooling) [inline src/app/cic/controllers/checkDetails.js:21]: Looks like we're missing some ; here. Zac has merged a PR with linter scripts, might want to give those a run?

File: src/app/cic/utils.test.js [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
+  it('returns a YYYY-MM-DD date with language set to cy as DD Month(Welsh) YYYY', () => {
+    expect(formatDate('1989-03-31', "cy")).to.equal('31 Mawrth 1989');
+  })
 
-  it("should return an empty string if date is empty string", () => {
-    expect(formatDate("", "YYYY-MM-DD")).to.equal("");
-  });
+  it('returns a YYYYMM/DD date as Invalid date', () => {
+    expect(formatDate('198903/31', "YYYY-MM-DD")).to.equal('Invalid Date');
+  })
```
Comment (Caitlin-cooling) [inline src/app/cic/utils.test.js:15]: hmm this means that we will be showing "Invalid date" to a user. I think where there is an error we actually probably want to show empty string? Maybe one to check with the BAs?

File: test/browser/step_definitions/CICCheckAnswersNavigation.step.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ Given(
     const cdPage = new CheckDetailsPage(await this.page);
 
     expect(await cdPage.isCurrentPage()).to.be.true;
-  }
+  },
```
Comment (Caitlin-cooling) [inline test/browser/step_definitions/CICCheckAnswersNavigation.step.js:17]: Hmm I don't think this formatting change is actually correct? What command did you run? Maybe @Z-Mohamoud can give us a hand?

Comment (GDainton) [reply test/browser/step_definitions/CICCheckAnswersNavigation.step.js:17]: I ran yarn lint and yarn prettier:fix

**Review by Caitlin-cooling — APPROVED** [review]

**Pull request merged**